### PR TITLE
additional improvements for "codepush release" related commands

### PR DESCRIPTION
**Pull request by sergey-akhalkov**
1. add missed `hasArg`s
2. add `projectRoot?` prop for `getCordovaProjectAppVersion` and `getReactNativeProjectAppVersion` methods
3. make all method in `file-utils.ts` synchronous
4. add platform validation both for React and Cordova
5. other small fixes and improvements

### CODE REVIEW

**Comment by buptkang** [comment]
@sergey-akhalkov I know you try to wrap up everything to improve. But I would suggest in the future, for each improvement, you can make a commit so that it would be easy for others to review. Otherwise, it is difficulty for me to read your mind. 😃

**Comment by buptkang** [comment]
LGTM. We can merge this and test it in the INT environment.

**Comment by sergey-akhalkov** [comment]
@buptkang, sure, thanks for the feedback, regarding the improvements list - please see comments from this PR: https://github.com/Microsoft/mobile-center-cli/pull/299

File: src/commands/codepush/lib/cordova-utils.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export function isValidOS(platform: string): boolean {
   }
 }
 
+export function isValidPlatform(platform: string): boolean {
+  return platform.toLowerCase() == "cordova";
```
Comment (christav) [inline src/commands/codepush/lib/cordova-utils.ts]: Should be `===`, not `==`

Comment (sergey-akhalkov) [reply src/commands/codepush/lib/cordova-utils.ts]: Got it, thanks for the feedback

**Pull request merged**